decidim-meetings/app/models/decidim/meetings/meeting.rb

```diff
@@ -298,6 +298,56 @@ def minutes_data?
         [video_url, audio_url].any?(&:present?)
       end
 
+      def self.sort_by_translated_title_asc
+        order(Arel::Nodes::InfixOperation.new("",
+                                              Arel::Nodes::InfixOperation.new("->>", arel_table[:title], Arel::Nodes.build_quoted(I18n.locale)), Arel.sql("ASC")))
+      end
+
+      def self.sort_by_translated_title_desc
+        order(Arel::Nodes::InfixOperation.new("",
+                                              Arel::Nodes::InfixOperation.new("->>", arel_table[:title], Arel::Nodes.build_quoted(I18n.locale)), Arel.sql("ASC")))
+      end
+
+      ransacker :type do
+        Arel.sql(%("decidim_meetings_meetings"."type_of_meeting"))
+      end
+
+      ransacker :title do
+        Arel.sql(%{cast("decidim_meetings_meetings"."title" as text)})
+      end
+
+      ransacker :id_string do
+        Arel.sql(%{cast("decidim_meetings_meetings"."id" as text)})
+      end
+
+      ransacker :is_upcomming do
+        query = <<-SQL.squish
+        (
+          SELECT EXISTS (
+            SELECT 1 FROM decidim_meetings_meetings as dmm WHERE dmm.id = decidim_meetings_meetings.id AND dmm.start_time > NOW()
+          )
+        )
+        SQL
+        Arel.sql(query)
+      end
+
+      ransacker :origin do
+        query = <<-SQL.squish
+        (
+        SELECT
+          CASE
+              WHEN decidim_author_type = 'Decidim::Organization' THEN 'official'
+              WHEN decidim_author_type = 'Decidim::UserBaseEntity' AND decidim_user_group_id IS NOT NULL THEN 'user_group'
+              WHEN decidim_author_type = 'Decidim::UserBaseEntity' AND decidim_user_group_id IS NULL THEN 'citizen'
+              ELSE 'unknown'
+          END
+        FROM decidim_meetings_meetings as dmm
+        WHERE dmm.id = decidim_meetings_meetings.id
+        )
+        SQL
+        Arel.sql(query)
```

Comment:
💬 Again, can we use 
```ruby
    query = <<-SQL.squish
      CASE
          WHEN decidim_author_type = 'Decidim::Organization' THEN 'official'
          WHEN decidim_author_type = 'Decidim::UserBaseEntity' AND decidim_user_group_id IS NOT NULL THEN 'user_group'
          WHEN decidim_author_type = 'Decidim::UserBaseEntity' AND decidim_user_group_id IS NULL THEN 'citizen'
          ELSE 'unknown'
      END
    SQL
    Arel.sql(query)
``` 
to avoid the extra query?

We are doing something similar [here](https://github.com/decidim/decidim/blob/fd99b8a13c503cafe421289005db373ed4b187bf/decidim-proposals/app/models/decidim/proposals/proposal.rb#L328).